src/PostRector/Rector/UnusedImportRemovingPostRector.php

```diff
@@ -55,25 +55,31 @@ public function enterNode(Node $node): ?Node
                 continue;
             }
 
-            $useUse = $namespaceStmt->uses[0];
             $isCaseSensitive = $namespaceStmt->type === Use_::TYPE_CONSTANT;
 
-            if ($isCaseSensitive) {
-                $names = $namesInOriginalCase;
-            } else {
-                if ($namesInLowerCase === null) {
-                    $namesInLowerCase = array_map(strtolower(...), $namesInOriginalCase);
+            $useCount = count($namespaceStmt->uses);
+            foreach ($namespaceStmt->uses as $useUseKey => $useUse) {
+                if ($isCaseSensitive) {
+                    $names = $namesInOriginalCase;
+                } else {
+                    if ($namesInLowerCase === null) {
+                        $namesInLowerCase = array_map(strtolower(...), $namesInOriginalCase);
+                    }
+
+                    $names = $namesInLowerCase;
                 }
 
-                $names = $namesInLowerCase;
-            }
+                if ($this->isUseImportUsed($useUse, $isCaseSensitive, $names, $namespaceName)) {
+                    continue;
+                }
+                if ($useCount > 1) {
+                    unset($namespaceStmt->uses[$useUseKey]);
```

Comment:
`$useCount` needs to be decrement until 0 for all uses removed:

```suggestion
                    unset($namespaceStmt->uses[$useUseKey]);
                    --$useCount;
```

For remove all uses:

```diff
namespace A;

-use B, C;

echo "test";

```

when multiple uses never used, not only partially.

Reply:
better way, just use unset by key, then remove the uses after loop if it empty array:

```php
// after loop
if ($hasChanged && $namespaceStmt->uses === []) {
    unset($node->stmts[$key]);
}
```

Reply:
I see your point. Done 👍🏽